Mathlib/NumberTheory/Padics/AddChar.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2025 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.NumberTheory.Padics.MahlerBasis
+
+/-!
+# Additive characters of `ℤ_[p]`
+
+We show that for any normed `ℤ_[p]`-algebra `R`, there is a canonical bijection between continuous
+additive characters `ℤ_[p] → R` and topologically nilpotent elements of `R`, given by sending `κ`
+to the element `κ 1 - 1`. This is used to define the Mahler transform for `p`-adic measures.
+
+Note that if the norm on `R` is not strictly multiplicative, then the condition that `κ 1 - 1` be
+topologically nilpotent is strictly weaker than assuming `‖κ 1 - 1‖ < 1`, although they are of
+course equivalent if `NormMulClass R` holds.
+-/
+
+open scoped fwdDiff
+open Filter Topology
+
+lemma fwdDiff_addChar_eq {M R : Type*} [AddCommMonoid M] [Ring R]
+    (φ : AddChar M R) (x h : M) (n : ℕ) : Δ_[h]^[n] φ x = (φ h - 1) ^ n * φ x := by
+  induction n generalizing x with
+  | zero => simp
+  | succ n IH =>
+    simp only [pow_succ, Function.iterate_succ_apply', fwdDiff, IH, ← mul_sub, mul_assoc]
+    rw [sub_mul, ← AddChar.map_add_eq_mul, add_comm h x, one_mul]
+
+variable {p : ℕ} [Fact p.Prime]
+
+/-- If two continuous additive characters of `ℤ_[p]` agree at 1, they agree everwhere. -/
+lemma AddChar.eq_of_apply_one_eq {M : Type*} [Monoid M] [TopologicalSpace M] [T2Space M]
+    {κ₁ κ₂ : AddChar ℤ_[p] M} (hκ₁ : Continuous κ₁) (hκ₂ : Continuous κ₂) (h : κ₁ 1 = κ₂ 1) :
+    κ₁ = κ₂ := by
+  refine DFunLike.coe_injective <| PadicInt.denseRange_natCast.equalizer hκ₁ hκ₂ (funext fun n ↦ ?_)
+  simp only [Function.comp_apply, ← nsmul_one, h, AddChar.map_nsmul_eq_pow]
+
+variable {R : Type*} [NormedRing R] [Algebra ℤ_[p] R] [IsBoundedSMul ℤ_[p] R]
+  [IsUltrametricDist R]
+
+lemma AddChar.tendsto_apply_one_sub_pow {κ : AddChar ℤ_[p] R} (hκ : Continuous κ) :
```

Comment:
Same question for the name: `apply` or `eval`?

Reply:
There are tons of `xxx_apply` lemmas in mathlib, I think this usage is pretty standard.

Reply:
I certainly agree, my feeling is that when we evaluate at "explicit" values we use `eval`  and `apply` is an abstract application, like in `RatFunc.eval_zero` vs `tprod_apply`. Not terribly important, though.